### PR TITLE
AIModuleWrapper now holds the original pointer instead of copying

### PR DESCRIPTION
**Pull request by 0x7CFE**

### CODE REVIEW

File: example/Dll.cpp [hunk 1]
```diff
@@ -86,11 +86,11 @@ extern "C"
         Broodwar = reinterpret_cast<Game*>(BroodwarPtr);
     }
     __declspec(dllexport) BWAPI::AIModule* newAIModule() {
-        ExampleModule module;
-        module.name = "ExampleModule";
-        module.vtable_ = &module_vtable;
+        ExampleModule* const module = new ExampleModule();
```
Comment (kpp) [inline example/Dll.cpp]: Use C, Luke.

Comment (dkashitsyn) [inline example/Dll.cpp]: Then why it's `.cpp` file?

Comment (kpp) [inline example/Dll.cpp]: It should be .c . I made a mistake

**Review by kpp — COMMENTED** [review]
Don't forget to fix AIModule.h docs

**Review by kpp — APPROVED** [review]

**Comment by 0x7CFE** [comment]
Let's merge it then

**Pull request merged**